### PR TITLE
Add client-side support for target_info endpoint

### PR DESCRIPTION
**Pull request by paaige**

### CODE REVIEW

File: qiskit-superstaq/qiskit_superstaq/superstaq_provider.py [hunk 1]
```diff
@@ -470,7 +470,10 @@ def cq_compile(
         )
 
     def supercheq(
-        self, files: List[List[int]], num_qubits: int, depth: int
+        self,
+        files: List[List[int]],
+        num_qubits: int,
+        depth: int,
     ) -> Tuple[List[qiskit.QuantumCircuit], npt.NDArray[np.float_]]:
         """Docstring."""
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/superstaq_provider.py:478]: lol (predates this pr)
```suggestion
        """Returns the randomly generated circuits and the fidelity matrix for inputted files."""
```

File: qiskit-superstaq/qiskit_superstaq/superstaq_provider.py [hunk 2]
```diff
@@ -470,7 +470,10 @@ def cq_compile(
         )
 
     def supercheq(
-        self, files: List[List[int]], num_qubits: int, depth: int
+        self,
+        files: List[List[int]],
+        num_qubits: int,
+        depth: int,
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/superstaq_provider.py:476]: why this change? if unnecessary,
```suggestion
        self, files: List[List[int]], num_qubits: int, depth: int
```

File: general-superstaq/general_superstaq/superstaq_client.py [hunk 3]
```diff
@@ -316,6 +317,24 @@ def supercheq(
         }
         return self.post_request("/supercheq", json_dict)
 
+    def target_info(
```
Comment (richrines1) [inline general-superstaq/general_superstaq/superstaq_client.py:320]: this will need its own test in `superstaq_client_test` to pass the standalone coverage check

**Review by richrines1 — COMMENTED** [review]
nice! mostly just nits, and `_SuperstaqClient.target_info` needs its own test

(also assume you've checked/will check this locally against the server endpoint?)

File: general-superstaq/general_superstaq/superstaq_client.py (hunk 3 above)
Comment (richrines1) [reply general-superstaq/general_superstaq/superstaq_client.py:320]: (also just noticed this bug as a result of this: #486)

**Comment by paaige** [comment]
> nice! mostly just nits, and `_SuperstaqClient.target_info` needs its own test
> 
> (also assume you've checked/will check this locally against the server endpoint?)

Yes, did check locally! But need to double check CSS (modified locally) before merging.

File: qiskit-superstaq/qiskit_superstaq/superstaq_provider.py (hunk 2 above)
Comment (paaige) [reply qiskit-superstaq/qiskit_superstaq/superstaq_provider.py:476]: This was  the work of the formatting script, will revert.

File: cirq-superstaq/cirq_superstaq/service.py [hunk 4]
```diff
@@ -676,7 +676,10 @@ def supercheq(
         fidelities = gss.serialization.deserialize(json_dict["fidelities"])
         return circuits, fidelities
 
-    def target_info(self, target: str,) -> Dict[str, Any]:
```
Comment (richrines1) [inline cirq-superstaq/cirq_superstaq/service.py]: the trailing comma after `str` here is why the formatter want this to be multiple lines; it should be fine with
```python
    def target_info(self, target: str) -> Dict[str, Any]:
```

File: qiskit-superstaq/qiskit_superstaq/superstaq_provider.py (hunk 2 above)
Comment (paaige) [reply qiskit-superstaq/qiskit_superstaq/superstaq_provider.py]: This still seems to be a problem (if I don't put the arguments on a new line, lint errors are thrown, looking into why this is the case now).

File: qiskit-superstaq/qiskit_superstaq/superstaq_provider.py [hunk 5]
```diff
@@ -472,10 +472,15 @@ def cq_compile(
     def supercheq(
         self, files: List[List[int]], num_qubits: int, depth: int
     ) -> Tuple[List[qiskit.QuantumCircuit], npt.NDArray[np.float_]]:
-        """Docstring."""
+        """Returns the randomly generated circuits and the fidelity matrix for inputted
+        files."""
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/superstaq_provider.py:476]: nit
```suggestion
        """Returns the randomly generated circuits and the fidelity matrix for inputted files."""
```

File: qiskit-superstaq/qiskit_superstaq/superstaq_backend.py [hunk 6]
```diff
@@ -136,3 +136,7 @@ def run(
         job = qss.SuperstaQJob(self, job_id)
 
         return job
+
+    def target_info(self) -> Any:
```
Comment (richrines1) [inline qiskit-superstaq/qiskit_superstaq/superstaq_backend.py:140]:
```suggestion
    def target_info(self) -> Dict[str, Any]::
```

File: general-superstaq/general_superstaq/superstaq_client.py [hunk 7]
```diff
@@ -316,6 +316,20 @@ def supercheq(
         }
         return self.post_request("/supercheq", json_dict)
 
+    def target_info(self, target: str) -> Any:
```
Comment (richrines1) [inline general-superstaq/general_superstaq/superstaq_client.py:319]:
```suggestion
    def target_info(self, target: str) -> Dict[str, Any]::
```

File: general-superstaq/general_superstaq/superstaq_client.py [hunk 8]
```diff
@@ -297,16 +297,16 @@ def supercheq(
...
-            circuit_return_type: Supports only `cirq` and `qiskit` for now
-
-        Returns: The output of SupercheQ
+            files: List of files specified as binary using ints.
+                For example: [[1, 0, 1], [1, 1, 1]].
+            num_qubits: Number of qubits to run SupercheQ on.
+            depth: The depth of the circuits to run SupercheQ on.
+            circuit_return_type: Supports only `cirq` and `qiskit` for now.
 
+        Returns: The output of SupercheQ.
```
Comment (richrines1) [inline general-superstaq/general_superstaq/superstaq_client.py:309]: nit
```suggestion
        Returns:
            The output of SupercheQ.
```

**Review by richrines1 — COMMENTED** [review]
(couple more nits and & annotations)

File: qiskit-superstaq/qiskit_superstaq/superstaq_provider.py (hunk 5 above)
Comment (richrines1) [reply qiskit-superstaq/qiskit_superstaq/superstaq_provider.py:476]: should still be corrected - (i think) this fits on a single line

**Review by richrines1 — APPROVED** [review]
one nit remaining from last review, otherwise lgtm!
